drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/loader/loader.rb

```diff
@@ -167,36 +167,46 @@ def importer_log
         next unless File.file?(source_file_path)
 
         encoding = AutoEncodingCsv.detect_encoding(source_file_path)
+        self.class.fix_bad_line_endings(source_file_path, encoding)
         File.open(source_file_path, 'r', encoding: encoding) do |file|
-          if bad_line_endings?(file)
-            copy_length = file.stat.size - 2
-            begin
-              logger.debug "Correcting bad line ending in #{source_file_path}"
-              tmp_file = ::Tempfile.new(file_name)
-              File.copy_stream(file, tmp_file, copy_length, 0)
-              tmp_file.write("\n")
-              File.open(tmp_file.path, 'r', encoding: encoding) do |tmp_file_io|
-                load_source_file_pg(read_from: tmp_file_io, klass: klass, original_file_path: source_file_path)
-              end
-            ensure
-              tmp_file.close
-              tmp_file.unlink
-            end
-          else
-            load_source_file_pg(read_from: file, klass: klass, original_file_path: source_file_path)
-          end
+          load_source_file_pg(read_from: file, klass: klass, original_file_path: source_file_path)
+        end
+      end
+    end
+
+    def self.fix_bad_line_endings(filename, encoding)
+      tmp_file = ::Tempfile.new(filename)
+      file_with_bad_line_endings = false
+
+      File.open(filename, 'r', encoding: encoding) do |file|
+        file_with_bad_line_endings = ! valid_line_endings?(file)
+      end
+      if file_with_bad_line_endings
+        File.open(filename, 'r', encoding: encoding) do |file|
+          copy_length = file.stat.size - 2
+          Rails.logger.debug "Correcting bad line ending in #{filename}"
+          File.copy_stream(file, tmp_file, copy_length, 0)
+          tmp_file.write("\n")
+          tmp_file.close
         end
+        FileUtils.cp(tmp_file, filename)
       end
+    ensure
+      tmp_file&.close
+      tmp_file&.unlink
```

Comment:
Not sure if the `&`'s are required -- but seem safer...

Reply:
They may not be, but in testing I hit a scenario where tmp_file was empty, so I left them.